src/App/DocumentObserver.h

```diff
@@ -28,10 +28,11 @@
 #include <Base/Bitmask.h>
 #include <boost_signals2.hpp>
 #include <memory>
-#include <set>
 #include <FCGlobal.h>
 
 
+// template<class T> class std::set;

```

Comment:
This is commented code. But as said above do not try to forward declare STL classes. For some classes the STL provide special headers like iosfwd for iostream classes but if there is no such header you have to include the regular file.

Reply:
@wwmayer, yes iosfwd is an example of forward declaration of iostream. However, I have had a very hard time to find forward declaration includes for std lib. Do you know where to find them? The closest I have gotten is https://blog.magnum.graphics/backstage/forward-declaring-stl-container-types/, that shows there are no standard for std lib forward declarations across compilers

Reply:
> Do you know where to find them?

No. iosfwd is the only official header of the C++ standard I know of. In https://en.cppreference.com/w/cpp/header no other header is listed that provides forward declarations.